PayPalMessaging/src/main/java/com/braintreepayments/api/PayPalMessagingView.kt

```diff
@@ -22,7 +22,16 @@ class PayPalMessagingView(
     private val braintreeClient: BraintreeClient,
     context: Context
 ) : FrameLayout(context) {
-    var payPalMessagingListener: PayPalMessagingListener? = null
+    private var listener: PayPalMessagingListener? = null
```

Comment:
Interesting I never realized that we make the listener value private. It's probably always been like this since the pre-Kotlin days.

We should consider moving to property-based access in `v5` to make the API more Kotlin friendly.

Reply:
Yeah I am down if we want to make that change across the board in v5. For now, just wanted to keep how this module sets the listener consistent with the other modules.

Reply:
Def yeah that's the right call. I only brought it up to make note.

Reply:
Oh for sure, it's also the more Swift-y approach which is almost certainly why I wrote it that way initially before looking at the other modules. 🙈

Reply:
Listeners are for the most part removed in v5, so I think with this module since we are kind of bridging v4 and v5 we can take whatever approach we want (assuming we keep the listener in v5 for this module only)

Reply:
Oooo interesting, would this be the more "kotlin" approach and `setListener` is the more java approach? I guess also is the a reason we wouldn't want both to exist or does the kotlin way get bridged somehow to a more java friendly interface? These may be super basic android questions. Lol.

Reply:
As far as Java v. Kotlin here, the __set__ prefix e.g. `setProperty()` is more of a traditional naming convention in Java. Kotlin actually has `get` and `set` baked [into the language](https://www.baeldung.com/kotlin/getters-setters#getters-and-setters-for-properties).

It's an important callout here too because the Java style getters and setters are a big source of boilerplate that bloats code size. The compiler will synthesize Java-style getters and setters for any `val` or `var` Kotlin properties, so anyone consuming our libs in Java won't have to know the underlying implementation is Kotlin-based.

We still need to codify our Kotlin and Java style guides, but I lean on Swift API Naming Guidelines and the JetBrains [Kotlin Style Guide](https://kotlinlang.org/docs/coding-conventions.html#idiomatic-use-of-language-features). Right now it's good to bend the rules until we figure out what feels right for us.

Reply:
Oh nice! Yeah if under the hood we get what we need, I am down to revert the `setListener` change and keep the dot syntax. Is that what you are thinking @sshropshire?

Reply:
TLDR from me, I think it's fine how you have it, but a good discussion to continue for `v5`.

Reply:
Cool cool 😅 - I'll keep it how it is in this PR for now and like you said as we solidify our style for v5 we can make updates.